bazelisk.sh

```diff
@@ -121,7 +121,7 @@ function main() {
         fi
     fi
 
-    case "$1" in
+    case "${1-}" in
```

Comment:
This looks clever, but I don't understand it! When I tried the following silly example to try it out, I couldn't avoid the unbound variable message
```sh
#!/bin/bash

set -eu

case "${1-}" in
    a*)
        echo aaa
        ;;
    *)
        echo nope
        ;;
esac
```

I get this:
```
rjs@sole:~$ bash foo.sh
foo.sh: line 6: 1: unbound variable
```

What am I missing?

Reply:
I think I must have been missing the hyphen(!) and not asking a sensible question. Please ignore! :-)

Reply:
`${parameter-word}` will expands to `${parameter}` if it's set, and `word` if it's not. This is similar to `${parameter:-word}`, which expands to `word` if `${parameter}` is not set *or empty*. Since the `word` here is empty anyway, we don't need the colon.

This is quite subtle in the bash doc. Grepping for `parameter-` gives you nothing. You have to grep for `parameter:-` and look at the small print.
> Omitting the colon results in a test only for a parameter that is unset.